Bumped Rails, Rspec-rails, Grape and Doorkeeper

texpert opened the pull request:
Rails 5.0
Rspec-rails 3.5
Grape 0.16.2
Doorkeeper 4.0

Also implemented Travis CI matrix for exclusion of Rails 5.0 testing with Ruby < 2.2.0

antek-drzewiecki commented on README.md:
```diff
@@ -29,7 +29,7 @@ Table of Contents
 ## Requirements
 - Ruby > 2.0
-- Doorkeeper > 1.4.0 and < 4.0
+- Doorkeeper > 1.4.0 and <= 4.0
```
This may be and < 5.0

texpert replied:
Let's be not so optimistic - < 4.1 will almost certainly work without problems, while further upgrade will need to be tested prior of announcing its support :)

antek-drzewiecki commented on wine_bouncer.gemspec:
```diff
@@ -18,16 +18,16 @@ Gem::Specification.new do |spec|
 
   spec.add_runtime_dependency 'grape', '~> 0.10', '< 1.0'
-  spec.add_runtime_dependency 'doorkeeper', '>= 1.4', '< 4.0'
+  spec.add_runtime_dependency 'doorkeeper', '>= 1.4', '<= 4.0'
```
Doorkeeper may be '< 5.0'

antek-drzewiecki commented on .travis.yml:
```diff
@@ -8,6 +8,10 @@ rvm:
 - 2.1.0
 - 2.2.0
+- 2.2.2
+- 2.2.3
```
Testing against 2.2.3 is not needed, otherwise the test matrix will become to large.

antek-drzewiecki commented on .travis.yml:
```diff
@@ -8,6 +8,10 @@ rvm:
 - 2.2.0
+- 2.2.2
+- 2.2.3
+- 2.2.4
```
Testing against 2.2.4 is not needed, otherwise the matrix will become to large

antek-drzewiecki commented on .travis.yml:
```diff
@@ -9,13 +9,15 @@ rvm:
 - 2.1.0
 - 2.2.0
 - 2.3.0
```
2.3.0 may be removed. ;)

antek-drzewiecki commented on Gemfile:
```diff
@@ -4,6 +4,8 @@ ENV['grape'] ||= '0.16.2'
 ENV['rails'] ||= '5.0.0'
 ENV['doorkeeper'] ||= '4.0.0'
 
+ruby '>=2.2.2' if ENV['rails'][0].to_i > 4
```
Not sure if this should be in the Gemfile. Rails will not run in ruby < 2.2.2 anyways.

texpert replied:
I think it is convenient when running bundle install with lower Ruby version to see the problem early.

antek-drzewiecki commented:
Hey Aurel, 

Thank you for the pull request. This is a nice effort you put in!
I've added some review comments. If you can take a look at them.

Before I can merge this request I need to make sure the Travis tests were running. 
No clue why Travis hasn't tested this build. But ill look in to it.

texpert commented:
Hm, maybe Travis remembered the tests from my local PR... Anyway, I will make a new commit addressing your comments and we'll see how it will go with Travis or no.

texpert commented:
Here it goes, @antek-drzewiecki ! :)

antek-drzewiecki commented:
Thank you, merged!
